tortoise/fields.py

```diff
@@ -375,7 +379,8 @@ class ForeignKeyField(Field):
                 Can only be set is field has a ``default`` set.
     """
 
-    __slots__ = ("model_name", "related_name", "on_delete")
+    # Here type will be set later, so we need a slot to be able to write it
+    __slots__ = ("type", "model_name", "related_name", "on_delete")
```

Comment:
Slotted instance variables and class variables cause conflicts. I have learnt to be very careful with this.
Consider injecting type as a class variable in the metaclass?
```py
attrs['type'] = bases[1] if len(bases) == 2 else bases[1:]
```
And then remove the `type = None` in the base class?

Reply:
It works.
I think you mean this:
```python
class A:
    __slots__ = ('x',)
    x = 1
```
Which doesn't work, because x is readonly. More here: https://stackoverflow.com/questions/5757924/python-slots-inheritance-and-class-variables-attribute-is-read-only-b

But this does:
```python
class A:
    x = 1

    
class B(A):
    __slots__ = ('x',)


a = A()
print(a.x)
a.x = 2
print(a.x)
```
And we use the latter here.